backend/app/main.py

```diff
@@ -33,7 +37,20 @@
 
 oauth2_scheme = HTTPBearer()
 
-### Decode JWT Token (Protect Routes)
+@app.exception_handler(Exception)
+async def catch_all_exceptions_handler(request: Request, exc: Exception):
+    """
+    Catch all exceptions and return a JSON response.
```

Comment:
This is a good pattern to use if we want to include more details in the original exception (such as stack traces and specific error details) but avoid divulging that information to the client. As it stands, however, this simply catches all  exceptions and repackages it to a 500-error response which is not what we want. I'm pretty sure this is what happens but let me know if you're testing shows otherwise.